test/e2e/mcd_test.go

```diff
@@ -425,6 +425,39 @@ func TestReconcileAfterBadMC(t *testing.T) {
 	}
 }
 
+func TestFullyDeleteMCShippedFiles(t *testing.T) {
+	cs := framework.NewClientSet("")
+	workerOldMcFirst := getMcName(t, cs, "worker")
+	mcHostFileFirst := createMCToAddFile("modify-host-file-first", "/etc/something", "mco-test", "root")
+	_, err := cs.MachineConfigs().Create(mcHostFileFirst)
+	require.Nil(t, err)
+	renderedConfig, err := waitForRenderedConfig(t, cs, "worker", mcHostFileFirst.Name)
+	require.Nil(t, err)
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", renderedConfig))
+	workerOldMcSecond := getMcName(t, cs, "worker")
+	mcHostFileSecond := createMCToAddFile("modify-host-file-second", "/etc/something", "mco-test-2", "root")
+	_, err = cs.MachineConfigs().Create(mcHostFileSecond)
+	require.Nil(t, err)
+	renderedConfig, err = waitForRenderedConfig(t, cs, "worker", mcHostFileSecond.Name)
+	require.Nil(t, err)
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", renderedConfig))
+	// now delete the MCs file, which must result in the file disappearing from the system
+	require.Nil(t, cs.MachineConfigs().Delete(mcHostFileSecond.Name, &metav1.DeleteOptions{}))
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", workerOldMcSecond))
+	require.Nil(t, cs.MachineConfigs().Delete(mcHostFileFirst.Name, &metav1.DeleteOptions{}))
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", workerOldMcFirst))
+	nodes, err := getNodesByRole(cs, "worker")
+	require.Nil(t, err)
+	for _, node := range nodes {
+		assert.Equal(t, node.Annotations[constants.CurrentMachineConfigAnnotationKey], workerOldMcFirst)
+		assert.Equal(t, node.Annotations[constants.MachineConfigDaemonStateAnnotationKey], constants.MachineConfigDaemonStateDone)
+		found := execCmdOnNode(t, cs, node, "cat", "/rootfs/etc/something")
```

Comment:
A stronger approach is something like `test -e /rootfs/etc/something` and checking the exit code.

Reply:
can do that thanks!